src/Rule/Consignment/ShipmentOptionsRule.php

```diff
@@ -9,24 +9,82 @@
+        }
+
+        parent::__construct();
+    }
+
     /**
      * @param  \MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment $validationSubject
      */
     public function validate($validationSubject): void
     {
         $options = [
-            AbstractConsignment::SHIPMENT_OPTION_AGE_CHECK      => $validationSubject->hasAgeCheck(),
-            AbstractConsignment::SHIPMENT_OPTION_INSURANCE      => $validationSubject->getInsurance(),
-            AbstractConsignment::SHIPMENT_OPTION_LARGE_FORMAT   => $validationSubject->isLargeFormat(),
-            AbstractConsignment::SHIPMENT_OPTION_ONLY_RECIPIENT => $validationSubject->isOnlyRecipient(),
-            AbstractConsignment::SHIPMENT_OPTION_RETURN         => $validationSubject->isReturn(),
-            AbstractConsignment::SHIPMENT_OPTION_SIGNATURE      => $validationSubject->isSignature(),
+            AbstractConsignment::SHIPMENT_OPTION_AGE_CHECK         => $validationSubject->hasAgeCheck(),
+            AbstractConsignment::SHIPMENT_OPTION_INSURANCE         => $validationSubject->getInsurance(),
+            AbstractConsignment::SHIPMENT_OPTION_LARGE_FORMAT      => $validationSubject->isLargeFormat(),
+            AbstractConsignment::SHIPMENT_OPTION_ONLY_RECIPIENT    => $validationSubject->isOnlyRecipient(),
+            AbstractConsignment::SHIPMENT_OPTION_RETURN            => $validationSubject->isReturn(),
+            AbstractConsignment::SHIPMENT_OPTION_SIGNATURE         => $validationSubject->isSignature(),
+            AbstractConsignment::SHIPMENT_OPTION_SAME_DAY_DELIVERY => $validationSubject->isSameDayDelivery(),
         ];
 
         foreach ($options as $option => $value) {
             if ($value && ! in_array($option, $validationSubject->getAllowedShipmentOptions(), true)) {
                 $this->addError($option . ' is not allowed in ' . get_class($validationSubject));
             }
         }
+
+        if (! empty($this->requiredOptions)) {
+            $this->validateRequired($options);
+        }
+
+        if (! empty($this->countrySpecific)) {
+            $this->validateCountrySpecific($validationSubject);
+        }
+    }
+
+    /**
+     * @param  \MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment $validationSubject
+     */
+    private function validateCountrySpecific(AbstractConsignment $validationSubject): void
+    {
+        foreach ($this->countrySpecific as $country => $conditionalOption) {
+            if ($country === $validationSubject->getCountry()) {
+                foreach ($conditionalOption as $option) {
+                    if ($option && ! in_array($option, $validationSubject->getAllowedShipmentOptions(), true)) {
+                        $this->addError($option . ' is not allowed in ' . get_class($validationSubject));
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param  array $options
+     */
+    private function validateRequired(array $options): void
```

Comment:
noem dit dan ook `validateRequiredOptions`, geen shortcuts nemen in namen van dingen, zorgt er alleen maar voor dat het lastig is om te lezen